### PR TITLE
Implement support for subdirectories in the make:controller command

### PR DESCRIPTION
**Pull request by merouanekhalili**

### CODE REVIEW

File: http/console/controller_make_command.go [hunk 1]
```diff
@@ -52,7 +55,10 @@ func (receiver *ControllerMakeCommand) getStub() string {
 
 // populateStub Populate the place-holders in the command stub.
 func (receiver *ControllerMakeCommand) populateStub(stub string, name string) string {
-	stub = strings.ReplaceAll(stub, "DummyController", str.Case2Camel(name))
+	controllerName, packageName, _ := parseName(name)
+
+	stub = strings.ReplaceAll(stub, "DummyController", str.Case2Camel(controllerName))
+	stub = strings.ReplaceAll(stub, "dummy_package", packageName)
```
Comment (hwbrzzl) [inline http/console/controller_make_command.go:61]:
```suggestion
	stub = strings.ReplaceAll(stub, "DummyPackage", packageName)
```

File: http/console/controller_make_command_test.go [hunk 2]
```diff
@@ -22,3 +22,31 @@ func TestControllerMakeCommand(t *testing.T) {
 	assert.True(t, file.Exists("app/http/controllers/users_controller.go"))
 	assert.True(t, file.Remove("app"))
 }
+
+func TestControllerWithSubFolderMakeCommand(t *testing.T) {
+	controllerMakeCommand := &ControllerMakeCommand{}
+	mockContext := &consolemocks.Context{}
+	mockContext.On("Argument", 0).Return("").Once()
+	err := controllerMakeCommand.Handle(mockContext)
+	assert.EqualError(t, err, "Not enough arguments (missing: name) ")
```
Comment (hwbrzzl) [inline http/console/controller_make_command_test.go:31]: We can remove this case that we had test it above.

File: http/console/controller_make_command.go [hunk 3]
```diff
@@ -61,5 +67,26 @@ func (receiver *ControllerMakeCommand) populateStub(stub string, name string) st
...
+	return pwd + "/app/http/controllers/" + folderPath + str.Camel2Case(controllerName) + ".go"
+}
+
+func parseName(name string) (string, string, string) {
+
+	parts := strings.Split(name, "/")
+
+	controllerName := parts[len(parts)-1]
+	packageName := defaultPackageName
+	filePath := strings.ToLower(strings.Join(parts[:len(parts)-1], "/"))
```
Comment (hwbrzzl) [inline http/console/controller_make_command.go:85]: I think we don't need to call strings.ToLower, maybe the user prefers the folder starting with uppercase ~

File: http/console/controller_make_command_test.go [hunk 4]
```diff
@@ -22,3 +22,31 @@ func TestControllerMakeCommand(t *testing.T) {
...
+}
+
+func TestControllerHasAndCorrectPackageNameMakeCommand(t *testing.T) {
+	controllerMakeCommand := &ControllerMakeCommand{}
+	mockContext := &consolemocks.Context{}
+	mockContext.On("Argument", 0).Return("").Once()
+	err := controllerMakeCommand.Handle(mockContext)
+	assert.EqualError(t, err, "Not enough arguments (missing: name) ")
+
+	mockContext.On("Argument", 0).Return("admin/UsersController").Once()
```
Comment (hwbrzzl) [inline http/console/controller_make_command_test.go:47]: Can you add a test case for `admin/login/UsersController`, please?

**Comment by DevHaoZi** [comment]
Another command `make:request` think should also support this feature.

**Comment by DevHaoZi** [comment]
Another command `make:request` think should also support this feature.

**Pull request closed without merging**